### ISSUE TITLE
cancelAllRequests not working?

### ISSUE THREAD

**Issue by kevinrenskers** [vadymmarkov/Malibu#119]
Super simpel test.

```
class ViewController: UIViewController {
  let networking = Networking<Endpoint>()

  override func viewDidLoad() {
    super.viewDidLoad()

    let promise = networking.request(.fetchUsers)

    promise.validate().done({ data in
      print("Success")
    }).fail({ error in
      print(error)
    })

    promise.cancel()
  }
}
```

With the method above, where I cancel a single network promise, the request is not made and neither the success or an error is printed. Great!

However, when I change that last line to `networking.cancelAllRequests()`, I do get "Success" printed out, and I can confirm that the request is indeed made.

**Comment by kevinrenskers** [comment]
Also, when I look at the code, I image that if the cancelling did work, the `.fail` handler would be called with a network error, right? Because it doesn't go through the Promise cancellation logic which suppresses the error by default.

**Comment by kevinrenskers** [comment]
Yeah, basically in the `ResponseHandler.handle` method this sort of check would need to be added I think:

```swift
if let error = error as? URLError, error.code == .cancelled {
  networkPromise.cancel()
  return
}
```

**Comment by kevinrenskers** [comment]
Update on the `cancelAllRequests` problem I am seeing: because `self.queue.addOperation(operation)` is itself done within an async operation (namely the `middlewarePromise`), by the time `cancelAllRequests` is executed, the queue is empty. Only after `cancelAllRequests` is run on the empty queue is the operation actually added to the queue. This seems like a problem to me :)

**Comment by vadymmarkov** [comment]
I think you're right about `cancelAllRequests` being called when the queue is still empty. I'm not sure how to fix it in a good way though. One option could be to store network promises in array and then iterate through it and cancel in `cancelAllRequests` function.
